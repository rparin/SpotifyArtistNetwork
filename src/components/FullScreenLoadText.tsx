import cn from "@/utils/cn";

export default function FullScreenLoadText(props: {
  text: string;
  classname?: string;
}) {
  return (
    <div className="absolute z-[110] flex max-h-screen w-full items-center justify-center text-center">
      <h1
        className={cn(
          "horizontal-mask select-none rounded-lg bg-teal-200/75 px-20 py-2 text-2xl backdrop-blur-md dark:bg-teal-600/60",
          props.classname
        )}>
        {props.text}
      </h1>
    </div>
  );
}
