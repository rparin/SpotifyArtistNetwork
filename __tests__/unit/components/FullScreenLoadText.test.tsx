import { render, screen } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import FullScreenLoadText from "@/components/FullScreenLoadText";

const PROPS = {
  text: "Loading...",
};

test("it renders component unchanged", () => {
  const { container } = render(<FullScreenLoadText {...PROPS} />);
  expect(container).toMatchSnapshot();
});

describe("testing Footer Accessibility", () => {
  expect.extend(toHaveNoViolations);

  test("it should have no accessibility violations", async () => {
    const { container } = render(<FullScreenLoadText {...PROPS} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

test("it should show the loading text", () => {
  render(<FullScreenLoadText {...PROPS} />);
  expect(screen.getByText(PROPS.text)).toBeInTheDocument();
});
