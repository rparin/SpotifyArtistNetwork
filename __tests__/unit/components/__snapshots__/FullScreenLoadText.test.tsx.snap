// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <div
    class="absolute z-[110] flex max-h-screen w-full items-center justify-center text-center"
  >
    <h1
      class="horizontal-mask select-none rounded-lg bg-teal-200/75 px-20 py-2 text-xl font-bold backdrop-blur-md dark:bg-teal-600/60"
    >
      Loading...
    </h1>
  </div>
</div>
`;
